### PR TITLE
refactor: Move `KeyboardManager` in `InputManager`

### PR DESCRIPTION
**Pull request by MartinZikmund**
GitHub Issue (If applicable): part of #8341 

## PR Type

What kind of change does this PR introduce?

- Refactoring (no functional changes, no api changes)

## What is the current behavior?

`KeyboardManager` has a different registration approach from pointers.

## What is the new behavior?

Both pointers and `KeyboardManager` have the same registration approach.

### <samp>🤖 Generated by Copilot at 8ba21d6</samp>

This pull request refactors the keyboard input handling for the Skia GTK and WPF backends, by introducing a common `IUnoKeyboardInputSource` interface and platform-specific implementations. This reduces code duplication, improves consistency, and simplifies the `CoreWindow` and `InputManager` classes. It also renames and moves some files and methods to better reflect their purpose and location.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/Uno.UI/UI/Xaml/Internal/InputManager.Keyboard.skia.cs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Input;
+
+namespace Uno.UI.Xaml.Core;
+
+partial class InputManager
+{
+	internal KeyboardManager Keyboard { get; private set; } = default!;
+
+	partial void ConstructKeyboardManager() => Keyboard = new(this);
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Internal/InputManager.Keyboard.skia.cs:14]: You can remove the `default!` and add `[MemberNotNull(nameof(Keyboard))]` to `CosntructKeyboardManager`.

File: src/Uno.UI/UI/Xaml/Internal/InputManager.Pointers.Managed.cs [hunk 2]
```diff
@@ -76,10 +76,10 @@ public void Init(object host)
 				throw new InvalidOperationException("Failed to initialize the PointerManager: cannot resolve the IUnoCorePointerInputSource.");
 			}
 
-			// Currently there is no need to filter on ContentRootType ('_inputManager._contentRoot.Type is ContentRootType.CoreWindow')
-			// As soon as we have a CoreWindow we configure it.
-			// This might be needed later once we support multi-windowing
```
Comment (Youssef1313) [inline src/Uno.UI/UI/Xaml/Internal/InputManager.Pointers.Managed.cs:81]: Is this comment unintentionally deleted?

Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Internal/InputManager.Pointers.Managed.cs:81]: Oh you are now checking ContentRootType. So the comment deletion is good.

**Review by Youssef1313 — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Internal/InputManager.Keyboard.skia.cs (hunk 1 above)
Comment (MartinZikmund) [reply src/Uno.UI/UI/Xaml/Internal/InputManager.Keyboard.skia.cs:14]: Applied the suggestions to the next PR (DragDrop manager)

**Pull request merged**